mesonbuild/build.py

```diff
@@ -1990,13 +2005,15 @@ def post_init(self) -> None:
         super().post_init()
         if 'cs' in self.compilers:
             raise InvalidArguments('Static libraries not supported for C#.')
-        if 'rust' in self.compilers:
+        if self.uses_rust():
             # If no crate type is specified, or it's the generic lib type, use rlib
-            if not hasattr(self, 'rust_crate_type') or self.rust_crate_type == 'lib':
+            if self.rust_crate_type == 'lib':
                 mlog.debug('Defaulting Rust static library target crate type to rlib')
                 self.rust_crate_type = 'rlib'
+            elif self.rust_crate_type == 'clib':
+                self.rust_crate_type = 'staticlib'
```

Comment:
For `lib` it's logging something about defaulting, maybe same here?

Reply:
done